docs/markdown/snippets/shared_static_only_args.md

```diff
@@ -0,0 +1,5 @@
+## `<lang>_(shared|static)_args` for both_library
+
+We now allow passing arguments like `c_static_args` and `c_shared_args`. This
+allows a `both_library` to have arguments specific to either the shared or
+static library, as well as common arguments to both.
```

Comment:
I think this doc (and those below) should be more specific on the intended usage. In my PR I had a more detailed example: https://github.com/mesonbuild/meson/pull/11501/files#diff-a568a81d6a181beacd53f50d513b2e6a7330a726c3fbdf6b7c56b0774cbb6558.

One important thing to document is that using those will cause both_libraries() to compile everything twice. It means it is recommended to use it only if absolutely necessary. We need to avoid people setting `c_static_args: '-DSTATIC_COMPILATION'` on non-Windows platforms.

Which.... makes the whole thing much more complicated, because that means you now should be using:
```meson
k = {}
if host_machine.system() == 'windows' and get_option('default_library') != 'shared'
  k = {'c_static_args': '-DSTATIC_COMPILATION'}
endif
library(..., kwargs: k)
```

Which... brings us back to using a dict... A better usage would be
```
library(..., c_static_args: is_windows ? ['-DSTATIC_COMPILATION'] : [])
```
If we make sure to allow, and document, that passing empty list does not cause duplicate compilation.

Reply:
We could also add the long awaited "none" kwarg value to remove that kwarg, e.g. `library(..., c_static_args: is_windows ? ['-DSTATIC_COMPILATION'] : none())`.

Reply:
I have patches to add a default(), but that depends on getting all of the typed_kwargs conversion done, because typed_kwargs knows what the default should be. I Can send those out as a draft later

Reply:
@dcbaker Note that in order to not block on getting `default()` we can also just document here that `[]` won't duplicate the build. Empty list is already often treated just like `default()` in many places.